lib/internal/bootstrap/node.js

```diff
@@ -67,12 +67,24 @@ if (ownsProcessState) {
 process.config = JSON.parse(internalBinding('native_module').config);
 
 const rawMethods = internalBinding('process_methods');
+
 // Set up methods and events on the process object for the main thread
 if (isMainThread) {
+  // Cache the working directory to prevent lots of lookups. If the working
+  // directory is changed by `chdir`, it'll be updated.
+  let cachedCwd = rawMethods.cwd();
+
   process.abort = rawMethods.abort;
   const wrapped = mainThreadSetup.wrapProcessMethods(rawMethods);
   process.umask = wrapped.umask;
-  process.chdir = wrapped.chdir;
+  process.chdir = (dir) => {
+    wrapped.chdir(dir);
+    // Update the cached working directory.
+    cachedCwd = rawMethods.cwd();
+  };
+  process.cwd = () => {
+    return cachedCwd;
+  };
 
```

Comment:
Instead of setting the `let cachedCwd` initially it could be `undefined`.
Then in `chdir` you could set `cachedCwd = void 0` or `undefined` (whichever node style digs).
Then in `cwd` you can check if `cachedCwd` is `undefined` and set it if it is with `cachedCwd = rawMethods.cwd()`.

This avoids the upfront setting of it initialization and avoids extra work in `chdir` and keeps the check within `cwd`.

Reply:
We call `process.cwd()` lots and lots of times during bootstrap. So initializing the value seems like a good idea to me.
I guess the undefined check won't make a huge difference but I personally don't think we have to make sure `process.chdir()` has to be fast (it should seldom be called).

Reply:
I changed it to an empty string instead of using `undefined` so it's monomorphic.